File: problemset-new/000/00035-search-insert-position/freedomDR.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * 每天写写二分查找，左闭右开标准，锻炼手部肌肉
 */
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int lo = 0, hi = nums.size();
        while(lo < hi) {
            int mid = lo + (hi-lo)/2;
            if(nums[mid] < target) lo = mid+1;
            else hi = mid;
        }
        return lo;
    }
};

int main()
{
    return 0;
}

File: problemset-new/007/00704-binary-search/freedomDR.go
package main

import "fmt"

func search(nums []int, target int) int {
	lo, hi := 0, len(nums)
	for lo < hi {
		mid := lo + (hi-lo)/2
		if nums[mid] < target {
			lo = mid + 1
		} else {
			hi = mid
		}
	}
	if lo == len(nums) || nums[lo] != target {
		return -1
	} else {
		return lo
	}
}
func main() {
	fmt.Println("vim-go")
}
